### PR TITLE
docs(rome_js_analyze): improve CONTRIBUTING and code gen

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

This PR improves the internal documentation for the analyzer in several ways:

- Improve the flow
- Fix title hierarchy issues
- Shorter sentences
- More consistent formatting
- Fix space issues

The PR also marginally improve the code generation for `just new-lintrule`.

## Test Plan

No change.

### CODE REVIEW

File: crates/rome_analyze/CONTRIBUTING.md [hunk 1]
```diff
@@ -4,210 +4,221 @@ The analyzer is a generic crate aimed to implement a visitor-like infrastructure
...
 
 1. Run the command
-	```shell
-	> just new-lintrule crates/rome_js_analyze/src/analyzers/nursery myRuleName
-	```
-	Rules go in different folders, and the folder depend on the type of query system your rule
-	will use:
+
+   ```shell
+   > just new-lintrule crates/rome_js_analyze/src/analyzers/nursery myRuleName
```
Comment (ematipico) [inline crates/rome_analyze/CONTRIBUTING.md:76]: One thing that could be improved is to have a link to how to install `just`. We take for granted that users can use it as it is.

File: crates/rome_analyze/CONTRIBUTING.md [hunk 2]
```diff
@@ -4,210 +4,221 @@ The analyzer is a generic crate aimed to implement a visitor-like infrastructure
...
+
+In addition to selecting a group, rules may be flagged as `recommended`.
+The **recommended rules** are enabled in the default configuration of the _Rome_ linter.
+As a general principle, a recommended rule should catch actual programming errors.
+For instance detecting a coding pattern that will throw an exception at runtime.
+Pedantic rules that check for certain unwanted patterns but may have high false positive rates,
+should be left off from the recommended set.
+Rules intended to be recommended should be flagged as such even if they are still part of the `nursery` group,
+as unstable rules are only enabled by default on unstable builds.
+This gives to the project time to test the rule, find edge cases, etc.
```
Comment (ematipico) [inline crates/rome_analyze/CONTRIBUTING.md:39]:
```suggestion
When implementing **new rules**, they have to be implemented under the group `nursery`.
New rules should always be considered unstable/not exhaustive.

In addition to selecting a group, rules may be flagged as `recommended`.
The **recommended rules** are enabled in the default configuration of the _Rome_ linter.
As a general principle, a recommended rule should catch actual programming errors.
For instance, detecting a coding pattern that will throw an exception at runtime.
Pedantic rules that check for specific unwanted patterns but may have high false positive rates
should be left off from the recommended set.
Rules intended to be recommended should be flagged as such even if they are still part of the `nursery` group,
as unstable rules are only enabled by default on unstable builds.
This gives the project time to test the rule, find edge cases, etc.
```

File: crates/rome_analyze/CONTRIBUTING.md [hunk 3]
```diff
@@ -4,210 +4,221 @@ The analyzer is a generic crate aimed to implement a visitor-like infrastructure
...
 
 ## Lint rules
 
-This gives to the project time to test the rule, find edge cases, etc.
+When creating or updating a lint rule, you need to be aware that there's a lot of generated code inside our toolchain.
+Our CI makes sure that this code is not out of sync and fails otherwise.
+See the [code generation section](#code-generation) for more details.
```
Comment (ematipico) [inline crates/rome_analyze/CONTRIBUTING.md:45]:
```suggestion
When creating or updating a lint rule, you need to be aware that there's a lot of generated code inside our toolchain.
Our CI ensures this code is not out of sync and fails otherwise.
See the [code generation section](#code-generation) for more details.
```

File: crates/rome_analyze/CONTRIBUTING.md [hunk 4]
```diff
@@ -4,210 +4,221 @@ The analyzer is a generic crate aimed to implement a visitor-like infrastructure
...
+       // .. code
+       fn diagnostic(_ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {}
+   }
+   ```
 
-Run the command `cargo test` and if you've done everything correctly, you should see some snapshots
-emitted with diagnostics and code actions.
+   While implementing the diagnostic, please keep [Rome's technical principals](https://rome.tools/#technical) in mind.
+   This function is called of every signal emitted by the `run` function, and it may return
+   zero or one diagnostic.
```
Comment (ematipico) [inline crates/rome_analyze/CONTRIBUTING.md:112]:
```suggestion
   While implementing the diagnostic, please keep [Rome's technical principals](https://rome.tools/#technical) in mind.
   This function is called for every signal emitted by the `run` function, and it may return
   zero or one diagnostic.
```

File: crates/rome_analyze/CONTRIBUTING.md [hunk 5]
```diff
@@ -4,210 +4,221 @@ The analyzer is a generic crate aimed to implement a visitor-like infrastructure
...
+       // .. code
+       fn action(_ctx: &RuleContext<Self>, _state: &Self::State) -> Option<JsRuleAction> {}
+   }
+   ```
 
-For simplicity, use [`just`](#using-just) to run all the commands with:
+   This function is called of every signal emitted by the `run` function.
+   It may return zero or one code action.
+   When returning a code action, you will need to pass `category` and `applicability` fields.
+   `category` must be `ActionCategory::QuickFix`, while `applicability` must be `Applicability:MaybeIncorrect`.
```
Comment (ematipico) [inline crates/rome_analyze/CONTRIBUTING.md:126]:
```suggestion
   This function is called for every signal emitted by the `run` function.
   It may return zero or one code action.
   When returning a code action, you must pass the `category` and `applicability` fields.
   `category` must be `ActionCategory::QuickFix`, while `applicability` must be `Applicability:MaybeIncorrect`.
```

`must be` seems to be a stretch; I mean, we have rules that  have `Applicability::Always`. Maybe we should set more context of **why** we _must_ set these values.

File: crates/rome_analyze/CONTRIBUTING.md [hunk 6]
```diff
@@ -4,293 +4,283 @@ The analyzer is a generic crate aimed to implement a visitor-like infrastructure
...
 
-Check our main [contribution document](https://github.com/rome/tools/blob/main/CONTRIBUTING.md#snapshot-tests)
-to know how to deal with the snapshot tests.
+   This function is called of every signal emitted by the `run` function.
+   It may return zero or one code action.
+   When returning a code action, you will need to pass `category` and `applicability` fields.
+   `category` must be `ActionCategory::QuickFix`, while `applicability` must be `Applicability:MaybeIncorrect`.
 
-### Code generation
+Don't forget to format your code with `cargo fmt` and lint with `cargo clippy`.
```
Comment (ematipico) [inline crates/rome_analyze/CONTRIBUTING.md:128]:
```suggestion
Don't forget to format your code with `cargo format` and lint with `cargo lint`.
```

Better use our internal aliases because we enforce more rules.

File: crates/rome_analyze/CONTRIBUTING.md [hunk 7]
```diff
@@ -4,293 +4,283 @@ The analyzer is a generic crate aimed to implement a visitor-like infrastructure
...
+```jsonc
+[
+	"var x = 1; foo(x);",
+	"for (var x of [1,2,3]) { foo(x); }",
+]
+```
 
-   When a rule's sole intention is to **mandate a single concept** - such as forcing the use of camel-casing - the rule should be named using the `use` prefix. For example, the rule to mandating the use of camel-cased variable names is named `useCamelCase`.
+Note that code in a file ending with the extension `.jsonc` are in a _script environment_.
+That's mean that you cannot use syntax that belongs to ECMAScript modules such as `import` and `export`.
```
Comment (ematipico) [inline crates/rome_analyze/CONTRIBUTING.md:166]: That's => This means that...
